go-controller/pkg/util/dpu_annotations.go

```diff
@@ -56,75 +59,119 @@ type DPUConnectionStatus struct {
 	}
+	annotations[DPUConnectionDetailsAnnot] = string(bytes)
 	return nil
 }
 
-func (scd *DPUConnectionDetails) AsAnnotation() (map[string]string, error) {
-	data, err := json.Marshal(scd)
-	if err != nil {
-		// We should not get here
-		return nil, fmt.Errorf("failed to set annotation. %v", err)
+// UnmarshalPodDPUConnDetails returns dpu connection details for the specified network
+func UnmarshalPodDPUConnDetails(annotations map[string]string, netName string) (*DPUConnectionDetails, error) {
+	ovnAnnotation, ok := annotations[DPUConnectionDetailsAnnot]
+	if !ok {
+		return nil, newAnnotationNotSetError("could not find OVN pod annotation in %v", annotations)
 	}
-	annot := make(map[string]string)
-	annot[DPUConnectionDetailsAnnot] = string(data)
-	return annot, nil
-}
 
-func (scs *DPUConnectionStatus) FromPodAnnotation(podAnnot map[string]string) error {
-	if annot, ok := podAnnot[DPUConnetionStatusAnnot]; ok {
-		if err := json.Unmarshal([]byte(annot), scs); err != nil {
-			return fmt.Errorf("failed to unmarshal DPUConnectionStatus. %v", err)
+	podScds := make(map[string]DPUConnectionDetails)
+	if err := json.Unmarshal([]byte(ovnAnnotation), &podScds); err != nil {
+		// legacy
+		if netName == types.DefaultNetworkName {
+			var scd DPUConnectionDetails
+			if err := json.Unmarshal([]byte(ovnAnnotation), &scd); err == nil {
+				return &scd, nil
+			}
 		}
-		return nil
+		return nil, fmt.Errorf("failed to unmarshal ovn pod annotation %q: %v",
+			ovnAnnotation, err)
 	}
-	return fmt.Errorf("failed to get DPUConnectionStatus pod annotation \"%s\" does not exist",
-		DPUConnetionStatusAnnot)
+	scd, ok := podScds[netName]
+	if !ok {
+		return nil, fmt.Errorf("no dpu connection details annotation for network %s: %q",
+			netName, ovnAnnotation)
+	}
+	return &scd, nil
 }
 
-func (scs *DPUConnectionStatus) SetPodAnnotation(podAnnotator kube.Annotator) error {
-	data, err := json.Marshal(scs)
-	if err != nil {
-		// We should not get here
-		return fmt.Errorf("failed to set annotation. %v", err)
+// MarshalPodDPUConnStatus returns a JSON-formatted annotation describing the pod's DPU connection status
+func MarshalPodDPUConnStatus(pannotations *map[string]string, scs *DPUConnectionStatus, netName string) error {
+	annotations := *pannotations
+	if annotations == nil {
+		annotations = make(map[string]string)
+		*pannotations = annotations
+	}
+	podScss := make(map[string]DPUConnectionStatus)
+	ovnAnnotation, ok := annotations[DPUConnetionStatusAnnot]
+	if ok {
+		// legacy scd annoation are not of different network
+		if err := json.Unmarshal([]byte(ovnAnnotation), &podScss); err != nil {
+			var legacyScs DPUConnectionStatus
+			if err := json.Unmarshal([]byte(ovnAnnotation), &legacyScs); err == nil {
+				podScss = map[string]DPUConnectionStatus{}
+				podScss[types.DefaultNetworkName] = legacyScs
+			} else {
+				return fmt.Errorf("failed to unmarshal ovn pod annotation %q: %v",
+					ovnAnnotation, err)
+			}
+		}
```

Comment:
can `UnmarshalPodDPUConnStatus` be reused here?

Reply:
same as above